build.gradle

```diff
@@ -1,25 +1,35 @@
+plugins {
+    id 'java'
+    id 'org.openjfx.javafxplugin' version '0.0.8'
+}
+
 group = 'com.gluonhq'
 version = '1.0.3-SNAPSHOT'
 
-apply plugin: 'java'
+repositories {
+    mavenCentral()
+}
 
-sourceCompatibility = 1.8
+sourceCompatibility = JavaVersion.VERSION_11
 
 subprojects {
+
     apply plugin: 'java'
     apply plugin: 'maven'
+    apply plugin: 'org.openjfx.javafxplugin'
     apply from: '../mavenPublish.gradle'
 
-    sourceCompatibility = 1.8
-
-    repositories {
-        mavenCentral()
-    }
+    sourceCompatibility = JavaVersion.VERSION_11
```

Comment:
Upgrading to Java 11 causes a compilation failure:

    guice/src/main/java/com/gluonhq/ignite/guice/GuiceContext.java:20: error: reference to Module is ambiguous
        private final Supplier<Collection<Module>> modules;
                                          ^
      both interface com.google.inject.Module in com.google.inject and class java.lang.Module in java.lang match

Reply:
Interesting, everything compiled when I was doing it. Will take a look..

Reply:
Found and fixed several issues.  
Added Micronaut sample, made sure all projects compile one by one and all samples run.